common/datavalues/src/types/data_type.rs

```diff
@@ -125,3 +129,29 @@ impl fmt::Display for DataType {
         write!(f, "{:?}", self)
     }
 }
+
+#[derive(
+    serde::Serialize, serde::Deserialize, Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord,
+)]
+pub enum IntervalUnit {
+    Year,
+    Month,
+    Day,
+    Hour,
+    Minute,
+    Second,
+}
+
+impl IntervalUnit {
+    pub fn avg_seconds(unit: IntervalUnit) -> i32 {
```

Comment:
For DateTime, seconds is ok, but maybe we should consider the following case:
```
SELECT
    toDate('2020-02-28') - toIntervalMonth(1) AS a,
    toDate('2020-03-28') - toIntervalMonth(1) AS b,
    toDate('2020-03-31') - toIntervalMonth(1) AS c

Query id: 4ae748f4-8914-48b7-80d7-1c54fe9cf1c8

┌──────────a─┬──────────b─┬──────────c─┐
│ 2020-01-28 │ 2020-02-28 │ 2020-02-29 │
└────────────┴────────────┴────────────┘
```

Maybe we should save the interval units and values

Reply:
I just tried, the query "select toDate('2020-02-28')" gives me this:

mysql> select toDate('2020-02-28');
+----------------------+
| toDate('2020-02-28') |
+----------------------+
| NULL                 |
+----------------------+
1 row in set (0.01 sec)



datafuse :) select toDate('2020-02-28');

┌─toDate('2020-02-28')─┐
│           1970-01-01 │
└──────────────────────┘


We might need an extra issue to address the toDate before this one.

Reply:
> We might need an extra issue to address the toDate before this one.

It's wip.

>  Notice that arrow also support Date32 and Date64, but we are using uint32 for Date.
`Date` in Datafuse/ClickHouse is u16 now, it's enough in our life.  Datafuse used our logical types other  than arrow's.
We must have `DataTypeInterval`, but it's a logical datatype,  seems arrow's interval type can meet our demand, I vote for using arrow's DataTypeInterval.

Take interval just as seconds is not enough, we must have the interval unit.

Eg:

`03-31  interval  -1 month`   ===> `02-29`  rather than `03-31` sub `30` days.

Reply:
HI @sundy-li , correct me if I am wrong. For the arithmetic PLUS operation of (DateTime + Interval),  since we don't have DataValue mapping to arrow's datetime, the underneath operation would be uint32, right ? So let's say if I map our Interval to arrow's interval,  we still can't leverage arrow's arithmetic operation, unless we change the mapping. Is my understanding correct ?

Reply:
Yes, but we can have our own `DateTime <op> Interval`. 

If we find  it's logical  `DateTime <op> Interval`, we can go to another specific path rather than the `u32 <op> i64`.

something like:
```
let datetime_col = col1.u32()?;
let interval_col = col2.i64()?;

let result = datetime_col.op_interval(&interval_col, time_unit, time_zone ...)?;
```

Reply:
Yes, you are right, we can't benefit if introducing array::interval in computing, we must write our own computing.
That's reasonable.


And there are many computing functions we should write our own. Such as `BinaryArray.like()`, if introducing charsets, the property of this method may like `BinaryArray.like<CharSet>()`.    
As someone suggested in tikv's like_impl: https://github.com/tikv/tikv/blob/master/components/tidb_query_expr/src/impl_like.rs

Reply:
@sundy-li, something I want to raise is, arrow interval::DateTime use struct([i32; 2]) as data value, while we don't have it in datavalue.  It is pretty hard to use arrow interval here unless we can also add a new data value.
Another option is like what you did,  use arrow::extension wrapping with ArrowDataType::Int32 for our interval.